spark-jobs/src/test/scala/filodb/downsampler/DownsamplerMainSpec.scala

```diff
@@ -553,4 +565,28 @@ class DownsamplerMainSpec extends FunSpec with Matchers with BeforeAndAfterAll w
     }
 
   }
+
+  it("should bring up DownsampledTimeSeriesShard and be able to read specific columns " +
+      "from gauge using MultiSchemaPartitionsExec") {
+
+    val downsampleTSStore = new DownsampledTimeSeriesStore(downsampleColStore, rawColStore,
+      DownsamplerSettings.filodbConfig)
+    downsampleTSStore.setup(BatchDownsampler.rawDatasetRef, DownsamplerSettings.filodbSettings.schemas,
+      0, rawDataStoreConfig, DownsamplerSettings.rawDatasetIngestionConfig.downsampleConfig)
+    downsampleTSStore.recoverIndex(BatchDownsampler.rawDatasetRef, 0).futureValue
+    val colFilters = seriesTags.map { case (t, v) => ColumnFilter(t.toString, Equals(v.toString)) }.toSeq
+    val queryFilters = colFilters :+ ColumnFilter("_metric_", Equals(gaugeName))
+    val exec = MultiSchemaPartitionsExec("someId", System.currentTimeMillis(),
+      1000, InProcessPlanDispatcher, BatchDownsampler.rawDatasetRef, 0, queryFilters, AllChunkScan,
+      colName = Option("sum"))
+    val queryConfig = new QueryConfig(DownsamplerSettings.filodbConfig.getConfig("query"))
+    val queryScheduler = Scheduler.fixedPool(s"$QuerySchedName", 3)
+    val res = exec.execute(downsampleTSStore, queryConfig)(queryScheduler, 1 minute)
+      .runAsync(queryScheduler).futureValue.asInstanceOf[QueryResult]
+    queryScheduler.shutdown()
+    res.result.size shouldEqual 1
+    res.result.head.rows.map(r => (r.getLong(0), r.getDouble(1))).toList shouldEqual
+      List((1574273042000L, 112.0))
```

Comment:
was only one sample returned ? Is this right ?

Reply:
@vishramachandran confirmed with Vish, this is correct